src/test/java/uk/gov/homeoffice/digital/sas/timecard/listeners/KafkaEntityListenerTest.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.homeoffice.digital.sas.timecard.listeners;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.homeoffice.digital.sas.timecard.enums.KafkaAction;
+import uk.gov.homeoffice.digital.sas.timecard.kafka.producers.KafkaProducerService;
+import uk.gov.homeoffice.digital.sas.timecard.model.TimeEntry;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.homeoffice.digital.sas.timecard.testutils.CommonUtils.getAsDate;
+import static uk.gov.homeoffice.digital.sas.timecard.testutils.TimeEntryFactory.createTimeEntry;
+
+@ExtendWith(MockitoExtension.class)
+class KafkaEntityListenerTest {
+
+  private final static UUID OWNER_ID = UUID.randomUUID();
+  private TimeEntry timeEntry;
+
+  @Mock
+  private KafkaProducerService<TimeEntry> kafkaProducerService;
+  private final TimeEntryKafkaEntityListener kafkaEntityListener =
```

Comment:
I do understand your need a concrete class for the constructor. For the declaration type, wouldn't it make more sense to it as `KafkaEntityListener` instead of `TimeEntryKafkaEntityListener`? as this test is for the parent abstract class

Reply:
Updated